### ISSUE TITLE
Inconsistet results on Debian

### ISSUE THREAD

**Issue by Valerio1474** [lpereira/hardinfo#87]
On my Debian/Sid system I used hardinfo to compare performances among different kernel versions on the same machine, but now benchmarking has become inconsistent: missing results, too much different values. See the attachments to have an idea.
[hf24042017.pdf](https://github.com/lpereira/hardinfo/files/1134577/hf24042017.pdf)
[hf10072017.pdf](https://github.com/lpereira/hardinfo/files/1134578/hf10072017.pdf)

**Comment by lpereira** [comment]
Since not much information has been provided, I'll have to guess a few things here.  Please feel free to correct me or provide more information that you think should help determine if this is indeed an issue or not, and if it is, to try and fix it.

You can't compare benchmark results from different versions of HardInfo, specially if they were built with different compilers.  Benchmarks are compiled with optimizations turned off, but even then, newer compilers might do things that older compilers didn't, so results will most likely vary.

Also, ensure that machines are using the performance CPU frequency governor.  Using anything such as ondemand will not help yield consistent results across runs.

**Comment by mckaygerhard** [comment]
that @lpereira said its true.. i used hardinfo compiled in debina lenny without any stupid gcc newer flags and loads correctly the benchmarking of some i3 cpu's.. but with hardinfo compiled in recent gcc's the results are very confusing..

inj the past some benchmarks was or have some proplems, of implementation.. but was solved!

**Comment by Valerio1474** [comment]
Ok, I undetstand that different versions can yeld different results,  so IMO there is no point in referencing the benchmark to other processors, because now in N-Queens, just for example, my processor a relatively new i7 4720Q is worse than an AMD Phenom II X4 940 that is really odd. In the same benchmark, it used to be under 1 second. But another strange thing is that many results are exactly 1 sec in time and the Fibonacci outcome is missing. I don't think it's just a matter of compiler.

**Comment by mckaygerhard** [comment]
no no, you miscunderstand.. differents **compilations** not versions of hardinfo.. the version always can be same or different of hardinfo.. but if the build environment (the producer of the program to run) specifically the compiler have some significant "enhanchements" or "special flags" those benchmarks does not return spected results...

**Comment by Valerio1474** [comment]
I still think there is no point in comparing results that are affected by compilers. Do you really pretend that the same compiler is used for a decade? So, if this is a variable, you are not benchmarking the processor but the compiler.

**Comment by mckaygerhard** [comment]
of course, the nature of procesor also affect.. i hear some time ago, AMD procesors manage floating point wors rather thant intel procesors.. now of course things changed.. 

in any case, if a compiler that have a decade works better, sorry for you, i use things that works.. not things that are bad and have bugs!

**Comment by lpereira** [comment]
I'll take a look at this.  There have been some changes in how benchmarks are executed since 0.5.1 that, in theory, shouldn't affect the results that much.  (They're now executed in a different process, but the benchmark code is the same.)

@mckaygerhard Please refrain from commenting when there's nothing to add to the conversation.

**Comment by mckaygerhard** [comment]
i tested with recent commits of July.. now the test in one of my i3 machines does not end.. only the graphic test was completed.

as curios thing.. i compile hardinfo in debian lenny and the graphic test and n-queens yet completed at least.. in debian lasted/sid only the graphic test was completed...

@lpereira, was a citation of a sense fact, i used older compiler due produced better much binaries rather than recent compiler.. companies and humans use that things that works.. only things that works.. if something dont work s¿i think there's no sense on use it due it cannot be used!

**Comment by mckaygerhard** [comment]
hi, tested in a g33 intel chipset, retunrs resuls very rare: (more power rather than xpected)

![rresultsbugribonachi](https://user-images.githubusercontent.com/1571724/37010542-6d0c4c90-20c2-11e8-979c-1189b2e5d0cc.png)
